File: Data Structures & Algorithms/house-robber/submission-1.cpp
#include<cstring>
class Solution {
public:
    int t[101];
    int solve (vector<int>& nums, int index) {
        if (index >= nums.size()) {
            return 0;
        }

        if (t[index] != -1) return t[index];

        int result = max(nums[index] + solve(nums, index+2), solve(nums, index+1));
        return t[index] = result;
    }
    int rob(vector<int>& nums) {
        memset(t, -1, sizeof(t));
        return solve(nums, 0);
    }
};

File: Data Structures & Algorithms/remove-element/submission-1.py
class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        new_size = 0;
        for i in range(len(nums)):
            if nums[i] != val:
                nums[new_size] = nums[i]
                new_size+=1
        return new_size

        